UefiPayloadPkg/Library/PciHostBridgeLib/PciHostBridgeLib.c

```diff
@@ -218,6 +219,47 @@ PciHostBridgeGetRootBridges (
   return ScanForRootBridges (Count);
 }
 
+/**
+  Return all the root bridge instances in an array.
+
+  @param Count  Return the count of root bridge instances.
+
+  @return All the root bridge instances in an array.
+          The array should be passed into PciHostBridgeFreeRootBridges()
+          when it's not used.
+**/
+PCI_SEGMENT_INFO *
+EFIAPI
+GetPciSegmentInfo (

```

Comment:
Current UPL is consuming below library, could we update existing code to support multi-segment?
UefiPayloadPkg\Library\PciSegmentInfoLibAcpiBoardInfo\PciSegmentInfoLibAcpiBoardInfo.c

For example:
GetPciSegmentInfo ()
{
If gUplPciSegmentInfoHobGuid HOB present
use this HOB to return data to caller.
else
keep existing code to use gUefiAcpiBoardInfoGuid HOB to return data to caller.
}

Reply:
Yes this was also throwing CI issue. fixed it.

Reply:
resolved. will discuss detail implementation separately.